commands/version/__tests__/version-command.test.js

```diff
@@ -502,6 +502,89 @@ describe("VersionCommand", () => {
     });
   });
 
+  describe("working on a detached HEAD", () => {
+    const detachedHEAD = async (fixture = "normal") => {
+      const cwd = await initFixture(fixture);
+      const sha = await execa.stdout("git", ["rev-parse", "HEAD"], { cwd });
+      await execa("git", ["checkout", sha], { cwd });
+      return cwd;
+    };
+
+    it("throws for version", async () => {
+      try {
+        const cwd = await detachedHEAD();
+        await lernaVersion(cwd)();
+      } catch (err) {
+        expect(err.prefix).toBe("ENOGIT");
+        expect(err.message).toBe("Detached git HEAD, please checkout a branch to choose versions.");
+      }
+
+      expect.assertions(2);
+    });
+
+    it("does not throw for version --force-publish --no-git-tag-version --no-changelog", async () => {
```

Comment:
In the event no tags are found, collect-updates acts as if you had passed `--force-publish` anyway.

Reply:
Ok, I guess this makes it much simpler. :)